src/main.rs

```diff
@@ -4,51 +4,102 @@ mod command;
 mod object;
 mod util;
 
-fn main() {
-    let args: Vec<String> = env::args().collect();
+#[derive(Debug, Clone)]
+enum Command {
+    Init,
+    Add {
+        files: Vec<String>, /* it's better to use PathBuf here! */
+    },
+    Commit {
+        message: String,
+    },
+    Branch {
+        delete: bool,
+        name: String,
+    },
+    Checkout {
+        branch: String,
+    },
+    Log,
+}
 
-    let command = &args[1];
+fn options() -> Command {
+    use bpaf::*;
 
-    if command == "init" {
-        command::init::init();
-        return;
-    }
+    let init = pure(Command::Init)
+        .to_options()
+        .command("init")
+        .help("Initialize new git repository");
+
+    let add = {
+        let files = positional("FILE").help("File to add").many();
+        construct!(Command::Add { files })
+            .to_options()
+            .command("add")
+            .help("Add a file")
+    };
+
+    let commit = {
+        let message = short('m')
+            .long("message")
+            .help("Commit changes")
+            .argument("MESSAGE");
+        construct!(Command::Commit { message })
+            .to_options()
+            .command("commit")
+            .help("Commit changes")
+    };
+
+    let branch = {
+        let delete = short('d')
+            .long("delete")
+            .help("Delete the branch?")
+            .switch();
+        let name = positional("BRANCH").help("Branch name");
+        construct!(Command::Branch { delete, name })
+            .to_options()
+            .command("branch")
+    };
+
+    let checkout = {
+        let branch = positional("BRANCH").help("Branch name");
+        construct!(Command::Checkout { branch })
+            .to_options()
+            .command("checkout")
+    };
+
+    let log = pure(Command::Log).to_options().command("log");
 
+    construct!([init, add, commit, branch, checkout, log])
+        .to_options()
+        .version(env!("CARGO_PKG_VERSION"))
+        .fallback_to_usage()
+        .run()
+}
+
+fn main() {
     if let Err(e) = util::path::find_git_root() {
         println!("{}", e);
         return;
     }
 
-    match command.as_str() {
-        "add" => {
-            let file_names = &args[2..];
-            command::add::add(file_names);
-        }
-        "commit" => match &args[2].to_string()[..] {
-            "-m" => {
-                let message = &args[3].to_string();
-                command::commit::commit(message.to_string());
-            }
-            _ => {
-                println!("no commit message");
-                return;
+    let args = options();
+    println!("Args: {args:?}");
+
+    match args {
+        Command::Init => command::init::init(),
```

Comment:
We don't explore `.git` directory when command is `init`, so I want you to execute `init` command checking before `find_git_root`.

Reply:
With aaba8707dbc1cf61f39ce39e3968beb0afd115d5 it should execute `find_git_root` only if command is not `Command:init`